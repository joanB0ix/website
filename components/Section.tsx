export interface SectionProps {
  title: string;
  children: React.ReactNode;
}

export function Section(props: SectionProps) {
  return (
    <div className="my-6">
      <h2 className="font-semibold pb-2">{props.title}</h2>
      {props.children}
    </div>
  );
}
