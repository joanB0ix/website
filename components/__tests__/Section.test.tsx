import { render, screen } from "@testing-library/react";
import { Section } from "../Section";

describe("Section", () => {
  const getTestView = () => {
    return (
      <Section title="Test Title">
        <p>Test Content</p>
      </Section>
    );
  };

  it("Should render", () => {
    render(getTestView());

    const headerElement = screen.getByText("Test Title");
    const paragraphElement = screen.getByText("Test Content");

    expect(headerElement).toBeInTheDocument();
    expect(headerElement.tagName).toBe("H2");
    expect(paragraphElement).toBeInTheDocument();
    expect(paragraphElement.tagName).toBe("P");
  });

  it("Should match snapshot", () => {
    const { asFragment } = render(getTestView());
    expect(asFragment()).toMatchSnapshot();
  });
});
