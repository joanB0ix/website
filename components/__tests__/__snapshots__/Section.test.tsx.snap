// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="my-6"
  >
    <h2
      class="font-semibold pb-2"
    >
      Test Title
    </h2>
    <p>
      Test Content
    </p>
  </div>
</DocumentFragment>
`;
